scripts/validate_new_tree_and_output_rdf.py

```diff
@@ -0,0 +1,735 @@
+                      parent_resource_uri,
+                      parent_label,
+                      parent_oncotree_code):
+    if not (parent_oncotree_code or (parent_resource_uri and parent_label)):
+        print(f"'{oncotree_code}' does not have a parent defined either by the '{graphite.CSV_PARENT_ONCOTREE_CODE}' or both '{graphite.CSV_PARENT_RESOURCE_URI}' and '{graphite.CSV_PARENT_LABEL}'")
+        sys.exit(1)
+
+def using_oncotree_code_to_define_parent(row):
+ return graphite.CSV_PARENT_ONCOTREE_CODE in row and row[graphite.CSV_PARENT_ONCOTREE_CODE]
+
+def is_tissue_node(row):
+    return row[graphite.CSV_ONCOTREE_CODE] == "TISSUE"
+
+def required_fields_defined(row, required_fields, csv_file):
+    row_required_fields = TISSUE_NODE_REQUIRED_FIELDS if is_tissue_node(row) else required_fields
+    for field in row_required_fields:
+        field_is_required(row[field], field, row[graphite.CSV_INTERNAL_ID], csv_file)
+
+def parent_is_defined_if_not_tissue(row):
+    if not is_tissue_node(row):
+        parent_oncotree_code = "" if graphite.CSV_PARENT_ONCOTREE_CODE not in row else row[graphite.CSV_PARENT_ONCOTREE_CODE]
+        parent_is_defined(row[graphite.CSV_ONCOTREE_CODE], row[graphite.CSV_PARENT_RESOURCE_URI], row[graphite.CSV_PARENT_LABEL], parent_oncotree_code)
+
+def validate_has_no_parents(row):
+    # the TISSUE node cannot have any parents set
+    if (row[graphite.CSV_PARENT_RESOURCE_URI] or
+        row[graphite.CSV_PARENT_LABEL] or
+        using_oncotree_code_to_define_parent(row)):
+        print(f"The '{graphite.CSV_LABEL}' node must not have any of these fields set: '{graphite.CSV_PARENT_RESOURCE_URI}', '{graphite.CSV_PARENT_LABEL}', '{graphite.CSV_PARENT_LABEL}' but at least one is in '{csv_file}'", file=sys.stderr)
+        sys.exit(1)
+
+def validate_csv_file(csv_file,
+                      expected_header,
+                      required_fields,
+                      resource_uri_to_internal_ids,
+                      oncotree_codes_to_internal_ids,
+                      resource_uri_to_labels):
+    # these fields are required and must be unique
+    resource_uri_set = set([])
+    internal_id_set = set([])
+    oncotree_code_set = set([])
+
+    with open(csv_file, 'r', encoding='utf-8-sig') as file:
+        reader = csv.DictReader(file)
+        # validate header
+        actual_header = reader.fieldnames
+        missing_fields = set(expected_header) - set(actual_header)
+        if missing_fields:
+            print(f"ERROR: missing the following expected fields from input file '{csv_file}': {', '.join(missing_fields)}", file=sys.stderr)
+            sys.exit(1)
+
+        label_mismatch_errors = []
+        parent_invalid_errors = []
+        for row in reader:
+            required_fields_defined(row, required_fields, csv_file)
+            parent_is_defined_if_not_tissue(row)
+
+            # check these columns are unique
+            field_is_unique(row[graphite.CSV_RESOURCE_URI],
```

Comment:
confused about this function, not sure I see how `resource_uri_set` is ever populated

also, the function itself doesn't seem to use the arguments being passed in. maybe misreading?